p2p/proto/protocols.md

```diff
@@ -1,11 +1,278 @@
+header.
+
+### Transactions and Receipts
+The transactions protocol is used to download the transactions and receipts in a range of blocks.
+
+Its name for negotiation is `/starknet/transactions/0.1.0-rc.0`
+
+Each single message is either a fin, or a [TransactionWithReceipt](./transaction.proto)
+(A pair of [Transaction](./transaction.proto) and [Receipt](./receipt.proto))
+
+Each transaction represents a Starknet transaction. For more detail on the different transaction
+types, their content and their hash calculation see [here](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/transactions/).
+
+A receipt is comprised of
+* The hash of the transaction this receipt belongs to
+* The actual fee paid for this transaction and the price unit used for the payment
+* The messages to l1 this transaction generated.
+* The execution resources this transaction took. Comprised of Cairo resources and data availability
+resources
+* The revert reason as a string if this transaction was reverted. If the `revert_reason` field is
+missing, it means that this transaction was successful
+
+The transactions and receipts need to be sent according to the transaction execution order.
+
+The commitment of the transactions in the header is comprised of the number of transactions and
+their merkle root.
+
+In order to verify the transactions, the node needs to calculate their hash and compare it to the
+transactions merkle root in the block header.
+
+The same is true for receipts. They also have a commitment in the header.
+
+The number can be used to delimiter between the transactions and receipts of each block. 
+
+In order to avoid potential DDoS, the querying peer should reject the connection and declare the
+replying peer malicious once it sent more transactions with receipts than the sum of transaction
+amounts in the headers of the requested blocks.
+
+### Events
+The Events protocol is used to download the events emitted in a range of blocks.
+
+Its name for negotiation is `/starknet/events/0.1.0-rc.0`
+
+Each single message is a fin or an [Event](./event.proto)
+
+Each event contains
+* The index of the transaction that emitted this event
+(If it's the first transaction in the block, the index will be 0).
+* The address of the contract that emitted this event.
+* The keys and data of this event.
+
+The commitment of the events in the header is comprised of the number of events and
+their merkle root.
+
+In order to verify the events, the node needs to calculate their hash and compare it to the
+events merkle root in the block header.
+
+The number can be used to delimiter between the events of each block, and the transaction index field
+of each event is used to delimiter between the events of each transaction within a block.
+
+**Important Note**: The transaction index is currently not part of the headers hash. Until this is
+fixed, the value of the `transaction_index` field in each event is trusted.
```

Comment:
Is this not implicit as part of the transaction commitment? The transactions must be ordered by index for the merkle tree commitment calculation to be correct?

Or are they ordered by transaction hash?